nodestream/project/pipeline_scope.py

```diff
@@ -26,26 +26,34 @@ def __init__(
         pipelines: List[PipelineDefinition],
         persist: bool = True,
         config: ScopeConfig = None,
-        targets: List[str] = None,
+        targets: List[str] = [],
```

Comment:
I think the usage is technically safe here, but I think we should avoid the pitfall of using [mutable default arguments](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments).